theano/sandbox/cuda/cuda_ndarray.cu

```diff
@@ -298,6 +298,25 @@ outstanding_mallocs(PyObject* self, PyObject * args)
     return PyInt_FromLong(_outstanding_mallocs[0]);
 }
 
+
+static void *work_mem = NULL;
+static size_t work_size = 0;
+
+/*
+ * Returns a chunk of memory for temporary work inside of an op. You can only
+ * request a single chunk of memory at a time since it is reused.
+ */
+void *get_work_mem(size_t sz) {
+    if (sz < work_size)
+        return work_mem;
+    device_free(work_mem);
```

Comment:
I admit I assumed that device_free handles NULL correctly, which seems to be the case since the initial value is NULL and the code works.

I'm not sure we should check the return value of device_free (and I'm not sure it should have a return value).  There is not much we can do if it fails because the only reason it can is if the context is ruined, and the malloc that follows will return the same error.